Sources/GoogleAI/FunctionCalling.swift

```diff
@@ -0,0 +1,235 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// A predicted function call returned from the model.
+public struct FunctionCall: Equatable, Encodable {
+  /// The name of the function to call.
+  public let name: String
+
+  /// The function parameters and values.
+  public let args: JSONObject
+}
+
+/// A `Schema` object allows the definition of input and output data types.
+///
+/// These types can be objects, but also primitives and arrays. Represents a select subset of an
+/// [OpenAPI 3.0 schema object](https://spec.openapis.org/oas/v3.0.3#schema).
+public class Schema: Encodable {
+  /// The data type.
+  let type: DataType
+
+  /// The format of the data.
+  let format: String?
+
+  /// A brief description of the parameter.
+  let description: String?
+
+  /// Indicates if the value may be null.
+  let nullable: Bool?
+
+  /// Possible values of the element of type ``DataType/string`` with "enum" format.
+  let enumValues: [String]?
+
+  /// Schema of the elements of type ``DataType/array``.
+  let items: Schema?
+
+  /// Properties of type ``DataType/object``.
+  let properties: [String: Schema]?
+
+  /// Required properties of type ``DataType/object``.
+  let requiredProperties: [String]?
+
+  enum CodingKeys: String, CodingKey {
+    case type
+    case format
+    case description
+    case nullable
+    case enumValues = "enum"
+    case items
+    case properties
+    case requiredProperties = "required"
+  }
+
+  /// Constructs a new `Schema`.
+  ///
+  /// - Parameters:
+  ///   - type: The data type.
+  ///   - format: The format of the data; used only for primitive datatypes.
+  ///     Supported formats:
+  ///     - ``DataType/integer``: int32, int64
```

Comment:
What does double backtick mean here? It doesn't seem to be a landing page title as documented at https://www.swift.org/documentation/docc/formatting-your-documentation-content.

Reply:
It's a link to the symbol: https://www.swift.org/documentation/docc/linking-to-symbols-and-other-content#Navigate-to-a-Symbol